### PR TITLE
Canonical order table

### PR DESCRIPTION
**Pull request by mulby**
@brianhw @jab5569 

There is a bug here where I have to run the task twice to get it to work.

### CODE REVIEW

File: edx/analytics/tasks/reports/orders_import.py [hunk 1]
```diff
@@ -84,8 +89,12 @@ def requires(self):
             'database': self.database,
         }
         yield (
+            # Oscar User Table
```
Comment (johnalbaker) [inline edx/analytics/tasks/reports/orders_import.py]: Otto

File: edx/analytics/tasks/reports/orders_import.py [hunk 2]
```diff
@@ -114,4 +123,188 @@ def requires(self):
...
+                -- exactly one of these joins should resolve.
+                LEFT OUTER JOIN shoppingcart_certificateitem ci ON ci.orderitem_ptr_id = oi.id
+                LEFT OUTER JOIN shoppingcart_paidcourseregistration pcr ON pcr.orderitem_ptr_id = oi.id
+                LEFT OUTER JOIN shoppingcart_courseregcodeitem crc ON crc.orderitem_ptr_id = oi.id
+                LEFT OUTER JOIN shoppingcart_donation d ON d.orderitem_ptr_id = oi.id
+
+                -- Ignore "cart", "defunct-cart" and "paying" statuses since they won't have corresponding transactions
+                WHERE oi.status IN ('purchased', 'refunded')
+            ) combined;
+        """
```
Comment (johnalbaker) [inline edx/analytics/tasks/reports/orders_import.py:338]: pretty bad ass.

**Comment by johnalbaker** [comment]
One nit, one comment, overall :+1: LGTM

**Comment by brianhw** [comment]
LGTM. :+1:

**Pull request merged**